cmd/iso_package/config.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/kdomanski/iso9660/util"
+)
+
+const (
+	tmpFolderPrefix = "iso_package_tmp"
+)
+
+type Config struct {
+	Source    string
+	Kickstart string
+	Arch      string
+
+	tmpFolder      string
+	isoSourceLabel string
+	dstFolderPath  string
+}
+
+// CleanAll remove all files from the temporal folder
+func (c *Config) CleanAll() {
+	os.RemoveAll(c.tmpFolder)
+}
+
+// Run add the given Kickstart file to  given iso
+func (c *Config) Run() (string, error) {
+	err := c.CheckConfig()
+	if err != nil {
+		return "", err
+	}
+
+	dirName, err := ioutil.TempDir("/tmp/", tmpFolderPrefix)
+	if err != nil {
+		return "", err
+	}
+	InfoLogger.Printf("Creating temp folder: %s", dirName)
+	c.tmpFolder = dirName
+
+	err = c.GetIso()
+	if err != nil {
+		return "", err
+	}
+
+	err = c.MountIso()
+	if err != nil {
+		return "", err
+	}
+
+	err = c.UpdateFiles()
+	if err != nil {
+		return "", err
+	}
+
+	path, err := c.CreateFinalIso()
+	if err != nil {
+		return "", err
+	}
+	return path, nil
+}
+
+// CheckConfig validates that current config struct is valid
+func (c *Config) CheckConfig() error {
+
+	if c.Source == "" {
+		return errors.New("Source file cannot be nil")
+	}
+
+	if c.Kickstart == "" {
+		return errors.New("kickstart file cannot be nil")
+	}
+
+	_, err := os.Stat(c.Kickstart)
+	if err != nil {
+		return fmt.Errorf("kickstart file cannot be found: %v", err)
+	}
+
+	// @todo dependencies programs here geniso, xorriso, blkid
+	return nil
+}
+
+// Mount the iso in the source_iso folder
+func (c *Config) MountIso() error {
+	cmd := execCommand("blkid -s LABEL -o value %s", c.getIsoFilepath())
```

Comment:
Does it OK to assume we have admin permission?

Reply:
admin permision is not needed

Reply:
@eloycoto @bardielle Can we resolve this?